TopMovies/MoviesConnector/CategoryVCConnector.swift

```diff
@@ -0,0 +1,122 @@
+//
+//  CategoryVCConnector.swift
+//  TopMovies
+//
+//  Created by Macbook Pro  on 17.02.2021.
+//
+
+import Foundation
+
+class CategoryVCConnector: BaseConnector<MoviesCategoryVCProps> {
+  var categoryId: MovieCategory.ID?
+  var downloadedMovies: [Movie.ID] = []
+  var isExistingMoviesConnected = false
+  
+  required init(updateProps: @escaping (MoviesCategoryVCProps) -> Void) {
+    super.init(updateProps: updateProps)
+  }
+  convenience init(categoryId: MovieCategory.ID, updateProps: @escaping (MoviesCategoryVCProps) -> Void) {
+    self.init(updateProps: updateProps)
+    self.categoryId = categoryId
+    mainStore.dispatch(RequestedMoviesListAction(categoryId: categoryId, requestType: .loadMore))
+  }
+  
+  override func newState(state: MainState) {
+    guard
+      let categoryId = categoryId,
+      let categoryName = state.movieCategoriesState.relational[categoryId]?.title,
+      let moviesList = state.categoriesPaginationState.paginated[categoryId]?.list
+    else { return }
+    
+    if !isExistingMoviesConnected {
+      let props = MoviesCategoryVCProps(
+        categoryId: categoryId,
+        categoryName: categoryName,
+        loadProgress: .init(isReloadInProgress: false,
+                            isLoadMoreInProgress: false),
+        moviesState: .init(hasNewData: true,
+                           moviesUsage: .reload,
+                           movies: moviesList
+                            .compactMap { MovieTableViewCellProps(movie:
+                                                                    state.moviesState.relational[$0]) }))
+      _updateProps(props)
+      isExistingMoviesConnected = !isExistingMoviesConnected
+      downloadedMovies = moviesList
+    }
+    let categoryState = state.categoriesPaginationState.paginated[categoryId]
+    if isDownloadingInProgress(categoryState)
+    {
+      var isReloadInProgress = false
+      var isLoadMoreInProgress = false
+      if case .downloading = categoryState?.reload {
+        isReloadInProgress = true
+      }
+      if case .downloading = categoryState?.loadMore {
+        isLoadMoreInProgress = true
+      }
```

Comment:
You do have prisms in the state, don't you?

`let isReloadInprogress = categoryState?.reload.isDownloading`